File: ABC-229/D.py
S = input()
k = int(input())


ok = 0
ng = len(S) + 1

A = [0]*(len(S)+1)
for i in range(len(S)):
    A[i+1] = A[i]
    if S[i]==".":
        A[i+1] += 1

def check(ans:int)->bool:
    """
    kこ書き換えて連続してansすることがあるか
    """
    for i in range(len(S)-ans+1):
        if A[i+ans] - A[i] <= k:
            return True
    return False





while ng-ok>1:
    m = (ok+ng)//2

    if check(m):
        ok = m
    else:
        ng = m

print(ok)

File: ABC-229/D.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

string S;
int k;
bool check(int ans) {
  for (int i = 0; i <= S.length() - ans; ++i) {
    int cnt_x = 0;
    for (int j = 0; j < ans; ++j) {
      if (S[i + j] == '.') {
        cnt_x++;
        if (cnt_x > k) {
          i += j ;
          break;
        }
      }
      if (j >= ans - 1)
        return true;
    }
  }
  return false;
}

int main() {
  cin >> S;
  cin >> k;
  int ok, ng;
  ok = 0;
  ng = S.length() + 1;

  while (ng - ok > 1) {

    int m = (ok + ng) / 2;
    if (check(m))
      ok = m;
    else
      ng = m;
  }

  cout << ok << endl;
  return 0;
}
